front/src/utils/date.ts

```diff
@@ -229,3 +229,18 @@ export function isArrivalDateInSearchTimeWindow(
   const arrivalDate = new Date(arrivalTime);
   return arrivalDate >= searchDatetimeWindow.begin && arrivalDate <= searchDatetimeWindow.end;
 }
+
+/**
+ * Generates an ISO arrival date string based on the provided date, hours, and minutes.
+ * @param {Date} date - The base date.
+ * @param {{ hours: number; minutes: number }} - An object containing the hours and the minutes of the arrival time.
+ * @returns {string} The ISO formatted arrival date string.
+ */
+export const generateISOArrival = (
+  date: Date,
+  { hours, minutes }: { hours: number; minutes: number }
```

Comment:
not really an issue, just a suggestion: we could create an explicit type instead of repeating those parameters